internal/archive/credentials_test.go

```diff
@@ -0,0 +1,262 @@
+package archive_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/canonical/chisel/internal/archive"
+)
+
+type matchTest struct {
+	url      string
+	err      string
+	username string
+	password string
+}
+
+type credentialsTest struct {
+	summary    string
+	credsFiles map[string]string
+	matchTests []matchTest
+}
+
+var credentialsTests = []credentialsTest{{
+	summary: "Parsing",
+	credsFiles: map[string]string{
+		"50test-logins": `
+machine example.netter login bar password foo
+machine example.net login foo password bar
+
+machine example.org:90 login apt password apt
+machine	example.org:8080
+login
+example	password 	 foobar
+
+machine example.org
+login anonymous
+password pass
+
+machine example.com/foo login user1 unknown token password pass1
+machine example.com/bar password pass2 login user2
+		unknown token
+machine example.com/user login user
+machine example.netter login unused password firstentry
+machine socks5h://example.last/debian login debian password rules
+`,
+	},
+	matchTests: []matchTest{
+		{"https://example.net/foo", "", "foo", "bar"},
+		{"https://user:pass@example.net/foo", "", "user", "pass"},
+		{"https://example.org:90/foo", "", "apt", "apt"},
+		{"https://example.org:8080/foo", "", "example", "foobar"},
+		{"https://example.net:42/foo", "", "foo", "bar"},
+		{"https://example.org/foo", "", "anonymous", "pass"},
+		{"https://example.com/apt", "", "", ""},
+		{"https://example.com/foo", "", "user1", "pass1"},
+		{"https://example.com/fooo", "", "user1", "pass1"},
```

Comment:
This one looks suspect. What's up here?

Reply:
An URL in a netrc file is a string prefix. The parser doesn't care about slashes. The linked Apt parser does the same: https://salsa.debian.org/apt-team/apt/-/blob/d9039b24/apt-pkg/contrib/netrc.cc#L96-111